src/user/rust.rs

```diff
@@ -0,0 +1,188 @@
+use parse_display::{Display, FromStr};
+use rust_nix_templater::{options::RustToolchainChannel, *};
+
+use crate::ActionStack;
+
+use super::{UserAction, UserMetadata, UserPrompt};
+
+#[derive(Debug, Clone, PartialEq, Eq, Display, FromStr)]
+pub(crate) enum Prompt {
+    #[display("generate flake")]
+    Generate,
+    #[display("set package name ({0})")]
+    SetPackageName(String),
+    #[display("set executable name ({0})")]
+    SetExecName(String),
+    #[display("set description ({0})")]
+    SetDescription(String),
+    #[display("set toolchain ({0})")]
+    SetToolchain(RustToolchainChannel),
+    #[display("set license ({0})")]
+    SetLicense(String),
+    #[display("toggle build outputs ({0})")]
+    ToggleBuildOutputs(bool),
+    #[display("toggle app outputs ({0})")]
+    ToggleAppOutputs(bool),
+    #[display("toggle library flag ({0})")]
+    ToggleLibrary(bool),
+    #[display("{0} channel")]
+    ChooseToolchain(RustToolchainChannel),
+    #[display("{0} license")]
+    ChooseLicense(String),
+}
+
+impl Into<UserPrompt> for Prompt {
+    fn into(self) -> UserPrompt {
+        UserPrompt::Rust(self)
+    }
+}
+
+impl Prompt {
+    pub(crate) fn process_prompt(
+        self,
```

Comment:
ok, it seems my assumption that we get an owned version of this usually is incorrect, so I think we should change it back to `&self`.